### PR TITLE
updated lronacal to compute sun to body distance from cached ephem data

### PR DESCRIPTION
**Pull request by Kelvinrr**
## Description

Added Spice:: sunToBodyDist() to get sun to body distance from sunPosition and bodyRotation. lronaccal attempts to use the camera model for computation else defaults to old implementation. 

 
## Related Issue

https://github.com/USGS-Astrogeology/ISIS3/issues/1790

## How Has This Been Tested?

Manually verified results. Need to write gtests for new sun-body distance function.  

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/base/objs/Spice/Spice.cpp [hunk 1]
```diff
@@ -90,7 +90,11 @@ namespace Isis {
    * @param noTables Indicates the use of tables.
    */
   Spice::Spice(Cube &cube, bool noTables) {  
-    init(cube, noTables);
+    init(*cube.label(), noTables);
+  }
+  
+  Spice::Spice(Pvl &lab, json isd) {
```
Comment (krlberry) [inline isis/src/base/objs/Spice/Spice.cpp:103]: Docstring on function? (Even though it's very straightforward)

File: isis/src/base/objs/Spice/Spice.cpp [hunk 2]
```diff
@@ -1338,6 +1339,21 @@ namespace Isis {
...
     return m_target->name();
   }
+  
+  
+  double Spice::sunToBodyDist() const {
+    std::vector<double> sunPosition = m_sunPosition->Coordinate();
+    std::vector<double> bodyRotation = m_bodyRotation->Matrix();
+    
+    for(size_t i = 0; i < bodyRotation.size(); i++) {
+      std::cout << bodyRotation[i] << std::endl;
```
Comment (krlberry) [inline isis/src/base/objs/Spice/Spice.cpp]: Debug cout statement to remove?

**Review by krlberry — DISMISSED** [review]
:+1: Looks good, just one documentation thing and a debug statement left in.

**Review by krlberry — APPROVED** [review]

**Review by scsides — APPROVED** [review]
Reasonable

**Pull request merged**